### PR TITLE
Update to Gradle 8.10

### PR DESCRIPTION
**Pull request by jamesfredley**

### CODE REVIEW

File: .github/workflows/groovy-joint-workflow.yml [hunk 1]
```diff
@@ -41,7 +41,7 @@ jobs:
         uses: actions/setup-java@v4
         with:
           distribution: 'adopt'
-          java-version: '11.0.6'
+          java-version: '17'
```
Comment (matrei) [inline .github/workflows/groovy-joint-workflow.yml:44]: Distribution should also be changed to `temurin`

File: build.gradle [hunk 2]
```diff
@@ -10,31 +10,25 @@ buildscript {
...
-    commonBuild = 'https://raw.githubusercontent.com/grails/grails-common-build/gradle7'
+
+    isSnapshot = project.projectVersion.endsWith('-SNAPSHOT')
+    isReleaseVersion = !isSnapshot
 }
 
-allprojects  {
+ext."signing.keyId" = project.hasProperty("signing.keyId") ? project.getProperty('signing.keyId') : System.getenv('SIGNING_KEY')
+ext."signing.secretKeyRingFile" = project.hasProperty("signing.secretKeyRingFile") ? project.getProperty('signing.secretKeyRingFile') : "${System.properties['user.home']}${File.separator}.gnupg${File.separator}secring.gpg"
+ext."signing.password" = project.hasProperty("signing.password") ? project.getProperty('signing.password') : System.getenv('SIGNING_PASSPHRASE')
```
Comment (matrei) [inline build.gradle]: These can be simplifed by uing `project.findProperty('x') ?: 'y'

File: build.gradle [hunk 3]
```diff
@@ -83,11 +77,45 @@ subprojects { project ->
         if (isGrailsPlugin) {
             apply plugin: "org.grails.grails-plugin"
         } else {
-            apply from: "${commonBuild}/common-project.gradle"
+            apply plugin: 'groovy'
+            apply plugin: 'java-library'
```
Comment (matrei) [inline build.gradle]: The `groovy` plugin is already applied on line 71 and the `java-library` plugin on line 76.

File: build.gradle [hunk 4]
```diff
@@ -104,7 +132,309 @@ subprojects { project ->
     tasks.withType(Test) {
         useJUnitPlatform()
     }
+
+    ext {
+        isExample = project.name.startsWith('example')
+        isGrailsPlugin = project.name.startsWith('grails-plugin')
+        pomInfo = {
+            delegate.name project.title
+            delegate.description subproject.projectDesc
```
Comment (matrei) [inline build.gradle:141]: `subproject` should probably be `project`

File: build.gradle [hunk 5]
```diff
@@ -104,7 +132,309 @@ subprojects { project ->
...
+                            url "https://repo.grails.org/grails/libs-snapshots-local"
+                        }
+                    }
+                }
+            }
+
+            publications {
+                maven(MavenPublication) {
+                    if (isGrailsPlugin) {
+                        artifactId(subproject.name - 'grails-plugin-')
```
Comment (matrei) [inline build.gradle:219]: `subproject` should probably be `project`

File: build.gradle [hunk 6]
```diff
@@ -104,7 +132,309 @@ subprojects { project ->
...
+                    } else if (project.name.contains('/')) {
+                        artifactId(project.name.substring(subproject.name.indexOf('/') + 1))
```
Comment (matrei) [inline build.gradle]: `subproject` should probably be `project`

File: build.gradle [hunk 7]
```diff
@@ -104,7 +132,309 @@ subprojects { project ->
...
+    }
+}
+
+tasks.withType(Groovydoc) {
+    group = DOCUMENTATION_GROUP
+    docTitle = "${project.title} - ${project.version}"
+    destinationDir = project.file("build/docs/api")
+    def files = []
+    project.rootProject.subprojects
+            .findAll { !it.name != 'docs' && !it.name.startsWith('examples') }
```
Comment (matrei) [inline build.gradle]: `!it.name != 'docs'` does not seem right

File: build.gradle [hunk 8]
```diff
@@ -104,7 +132,309 @@ subprojects { project ->
...
+    asciidoc = true
+    resourcesDir = project.file(docResourcesDir)
+    properties = ['safe'     : 'UNSAFE',
+                  'version'  : project.version,
+                  'subtitle' : project.projectDesc,
+                  'api'      : '../api',
+                  'sourceDir': rootProject.projectDir.absolutePath,
+                  'sourcedir': rootProject.projectDir.absolutePath,
+                  'javaee'   : 'https://docs.oracle.com/javaee/7/api/',
+                  'javase'   : 'https://docs.oracle.com/javase/7/docs/api/',
```
Comment (matrei) [inline build.gradle]: These should probably be updated at some point

File: .github/workflows/groovy-joint-workflow.yml (hunk 1 above)
Comment (jamesfredley) [reply .github/workflows/groovy-joint-workflow.yml:44]: Updated

File: build.gradle (hunk 2 above)
Comment (jamesfredley) [reply build.gradle]: Updated

File: build.gradle (hunk 3 above)
Comment (jamesfredley) [reply build.gradle]: Updated

File: build.gradle (hunk 4 above)
Comment (jamesfredley) [reply build.gradle:141]: There were three subproject sections after removing commonBuild.  I had consolidated them, but went back and kept the first two separate to address.  

subprojects { subproject ->

subprojects { project ->

subproject {

File: build.gradle (hunk 7 above)
Comment (jamesfredley) [reply build.gradle]: Updated.  Certainly an advantage of removing the "invisible" commonBuild approach.  

matches the way it is done in gorm-hibernate5
https://github.com/grails/gorm-hibernate5/blob/0f3e6ed84a3c552de023c4ec278bc663548f4bf1/docs/build.gradle#L48

File: build.gradle (hunk 8 above)
Comment (jamesfredley) [reply build.gradle]: Updated

**Comment by jamesfredley** [comment]
The same adjustments were made on https://github.com/grails/grails-gsp/pull/478

except a few apply plugin lines, which were required.

**Comment by jamesfredley** [comment]
The build still needs a lot of polish at some later date and this is after substantial consolidation of what was in the commonBuild which overlapped and conflicted materially. It is much better now that we see it all directly instead of being applied from files on https://raw.githubusercontent.com/grails/grails-common-build/gradle7

File: build.gradle [hunk 9]
```diff
@@ -10,31 +10,25 @@ buildscript {
...
-    commonBuild = 'https://raw.githubusercontent.com/grails/grails-common-build/gradle7'
+
+    isSnapshot = project.projectVersion.endsWith('-SNAPSHOT')
+    isReleaseVersion = !isSnapshot
 }
 
-allprojects  {
+ext."signing.keyId" = project.hasProperty("signing.keyId") ?: System.getenv('SIGNING_KEY')
+ext."signing.secretKeyRingFile" = project.hasProperty("signing.secretKeyRingFile") ?: "${System.properties['user.home']}${File.separator}.gnupg${File.separator}secring.gpg"
+ext."signing.password" = project.hasProperty("signing.password") ?: System.getenv('SIGNING_PASSPHRASE')
```
Comment (matrei) [inline build.gradle]: Change `project.hasProperty('x')` to `project.findProperty('x')`

File: build.gradle [hunk 10]
```diff
@@ -106,5 +132,310 @@ subprojects { project ->
...
+        }
+        tasks.findByName("assemble").finalizedBy(sourcesJar, javadocJar)
+
+        publishing {
+            if (isSnapshot) {
+                repositories {
+                    maven {
+                        credentials {
+                            def u = System.getenv("ARTIFACTORY_USERNAME") ?: subproject.hasProperty("artifactoryPublishUsername") ? subproject.artifactoryPublishUsername : ''
+                            def p = System.getenv("ARTIFACTORY_PASSWORD") ?: subproject.hasProperty("artifactoryPublishPassword") ? subproject.artifactoryPublishPassword : ''
```
Comment (matrei) [inline build.gradle]: Change `subproject.hasProperty('x') ? subproject.x : ''` to `project.findProperty('x') ?: ''`

File: build.gradle [hunk 11]
```diff
@@ -106,5 +132,310 @@ subprojects { project ->
...
+if (isReleaseVersion) {
+    apply plugin: 'maven-publish'
+    apply plugin: "io.github.gradle-nexus.publish-plugin"
+
+    nexusPublishing {
+        repositories {
+            sonatype {
+                def ossUser = System.getenv("SONATYPE_USERNAME") ?: project.hasProperty("sonatypeOssUsername") ? project.sonatypeOssUsername : ''
+                def ossPass = System.getenv("SONATYPE_PASSWORD") ?: project.hasProperty("sonatypeOssPassword") ? project.sonatypeOssPassword : ''
+                def ossStagingProfileId = System.getenv("SONATYPE_STAGING_PROFILE_ID") ?: project.hasProperty("sonatypeOssStagingProfileId") ? project.sonatypeOssStagingProfileId : ''
```
Comment (matrei) [inline build.gradle]: Change `project.hasProperty('x') ? project.x : ''` to `project.findProperty('x') ?: ''`

File: build.gradle (hunk 9 above)
Comment (jamesfredley) [reply build.gradle]: Updated

File: build.gradle (hunk 10 above)
Comment (jamesfredley) [reply build.gradle]: Updated

File: build.gradle (hunk 11 above)
Comment (jamesfredley) [reply build.gradle]: Updated

**Review by matrei — APPROVED** [review]

**Pull request merged**